src/main/java/sg/ncl/DataController.java

```diff
@@ -188,6 +205,129 @@ public String getPublicDatasets(Model model) {
         return "data_public";
     }
 
+    @RequestMapping("{datasetId}/resources")
+    public String getResources(Model model, @PathVariable String datasetId, HttpSession session, RedirectAttributes redirectAttributes) {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        ResponseEntity response = restTemplate.exchange(properties.getDataset(datasetId), HttpMethod.GET, request, String.class);
+        String dataResponseBody = response.getBody().toString();
+        JSONObject dataInfoObject = new JSONObject(dataResponseBody);
+        Dataset dataset = extractDataInfo(dataInfoObject.toString());
+        if (!dataset.getContributorId().equals(session.getAttribute("id").toString())) {
+            log.warn(UPLOAD_DISALLOWED);
+            redirectAttributes.addFlashAttribute(MESSAGE_ATTRIBUTE, UPLOAD_DISALLOWED);
+            return "redirect:/data";
+        }
+        model.addAttribute("dataset", dataset);
+        return "data_resources";
+    }
+
+    /**
+     * References:
+     * [1] https://github.com/23/resumable.js/blob/master/samples/java/src/main/java/resumable/js/upload/UploadServlet.java
+     */
+    @RequestMapping(value="{datasetId}/resources/upload", method=RequestMethod.GET)
+    public ResponseEntity<String> checkChunk(@PathVariable String datasetId, HttpServletRequest request) {
+        int resumableChunkNumber = getResumableChunkNumber(request);
+        ResumableInfo info = getResumableInfo(request);
+
+        String url = properties.checkUploadChunk(datasetId, resumableChunkNumber, info.resumableIdentifier);
+        log.debug("URL: {}", url);
+        HttpEntity<String> httpEntity = createHttpEntityHeaderOnly();
+        ResponseEntity responseEntity = restTemplate.exchange(url, HttpMethod.GET, httpEntity, String.class);
+        String body = responseEntity.getBody().toString();
+        log.debug(body);
+        if (body.equals("Not found")) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(body);
+        }
+        return ResponseEntity.ok(body);
+    }
+
+    @RequestMapping(value="{datasetId}/resources/upload", method=RequestMethod.POST)
+    public ResponseEntity<String> uploadChunk(@PathVariable String datasetId, HttpServletRequest request) {
+        int resumableChunkNumber = getResumableChunkNumber(request);
+        ResumableInfo info = getResumableInfo(request);
+
+        try {
+            InputStream is = request.getInputStream();
+            ByteArrayOutputStream os = new ByteArrayOutputStream();
+            long readed = 0;
+            long content_length = request.getContentLength();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00117)